src/runtime/runtime.cs

```diff
@@ -360,7 +365,7 @@ internal static void Shutdown()
                 {
                     // Some clr runtime didn't implement GC.WaitForFullGCComplete yet.
                 }
-                PyEval_SaveThread();
+                PyGILState_Release(state);
```

Comment:
Just keep using `PyEval_SaveThread`, you need to release the GIL definitely here.

Reply:
I thought we needed to keep the return value of `PyGILState_Release`. Took a look at cpython's source code, we do not.